flax/linen/linear.py

```diff
@@ -282,6 +283,128 @@ def __call__(self, inputs: Array) -> Array:
     return y
 
 
+class Einsum(Module):
+  """An einsum transformation with learnable kernel and bias.
+
+  Example usage::
+
+    >>> import flax.linen as nn
+    >>> import jax, jax.numpy as jnp
+
+    >>> layer = nn.Einsum((5, 6, 7), 'abc,cde->abde')
+    >>> params = layer.init(jax.random.key(0), jnp.ones((3, 4, 5)))
+    >>> jax.tree_map(jnp.shape, params)
```

Comment:
use canonical `variables` name
```suggestion
    >>> variables = layer.init(jax.random.key(0), jnp.ones((3, 4, 5)))
    >>> jax.tree_map(jnp.shape, variables)
```